sse2neon.h

```diff
@@ -8358,6 +8359,549 @@ FORCE_INLINE int _mm_testz_si128(__m128i a, __m128i b)
+        PCMPSTR_EQ(a, b, mtx, size, number_of_lanes);                  \
+        return aggregate_equal_ordered_##size##x##number_of_lanes(     \
+            number_of_lanes, la, lb, mtx);                             \
+    }
+
+#define PCMPSTR_EQ(a, b, mtx, size, number_of_lanes)                 \
+    do {                                                             \
+        SSE2NEON_EVAL(SSE2NEON_REPEAT(number_of_lanes, EQ, u##size)) \
+    } while (0)
+
+#if !defined(__aarch64__)
+/* emulate vaddv u8 variant */
+static inline uint8_t vaddv_u8(uint8x8_t v8)
+{
+    const uint64x1_t v1 = vpaddl_u32(vpaddl_u16(vpaddl_u8(v8)));
+    return vget_lane_u8(vreinterpret_u8_u64(v1), 0);
+}
+
+/* emulate vaddvq u8 variant */
+static inline uint8_t vaddvq_u8(uint8x16_t a)
+{
+    uint8x8_t tmp = vpadd_u8(vget_low_u8(a), vget_high_u8(a));
+    uint8_t res = 0;
+    for (int i = 0; i < 8; ++i)
+        res += tmp[i];
+    return res;
+}
+
+/* emulate vaddvq u16 variant */
+static inline uint16_t vaddvq_u16(uint16x8_t a)
+{
+    uint32x4_t m = vpaddlq_u16(a);
+    uint64x2_t n = vpaddlq_u32(m);
+    uint64x1_t o = vget_low_u64(n) + vget_high_u64(n);
+
+    return vget_lane_u32((uint32x2_t) o, 0);
+}
+#endif
+
+static int aggregate_equal_any_8x16(int la, int lb, __m128i mtx[16])
+{
+    int res = 0;
+    int m = (1 << la) - 1;
+    uint8x8_t vec_mask = vld1_u8(_sse2neon_cmpestr_mask8b);
+    uint8x8_t t_lo = vtst_u8(vdup_n_u8(m & 0xff), vec_mask);
+    uint8x8_t t_hi = vtst_u8(vdup_n_u8(m >> 8), vec_mask);
+    uint8x16_t vec = vcombine_u8(t_lo, t_hi);
+    for (int j = 0; j < lb; j++) {
+        mtx[j] = vreinterpretq_m128i_u8(
+            vandq_u8(vec, vreinterpretq_u8_m128i(mtx[j])));
+        mtx[j] = vreinterpretq_m128i_u8(
+            vshrq_n_u8(vreinterpretq_u8_m128i(mtx[j]), 7));
+        int tmp = vaddvq_u8(vreinterpretq_u8_m128i(mtx[j])) ? 1 : 0;
+        res |= (tmp << j);
+    }
+    return res;
+}
+
+static int aggregate_equal_any_16x8(int la, int lb, __m128i mtx[16])
+{
+    int res = 0;
+    int m = (1 << la) - 1;
+    uint16x8_t vec =
+        vtstq_u16(vdupq_n_u16(m), vld1q_u16(_sse2neon_cmpestr_mask16b));
+    for (int j = 0; j < lb; j++) {
+        mtx[j] = vreinterpretq_m128i_u16(
+            vandq_u16(vec, vreinterpretq_u16_m128i(mtx[j])));
+        mtx[j] = vreinterpretq_m128i_u16(
+            vshrq_n_u16(vreinterpretq_u16_m128i(mtx[j]), 15));
+        int tmp = vaddvq_u16(vreinterpretq_u16_m128i(mtx[j])) ? 1 : 0;
+        res |= (tmp << j);
+    }
+    return res;
+}
+
+static int _sse2neon_cmp_byte_equal_any(__m128i a, int la, __m128i b, int lb)
+{
+    __m128i mtx[16];
+    PCMPSTR_EQ(a, b, mtx, 8, 16);
+    return aggregate_equal_any_8x16(la, lb, mtx);
+}
+
+static int _sse2neon_cmp_word_equal_any(__m128i a, int la, __m128i b, int lb)
+{
+    __m128i mtx[16];
+    PCMPSTR_EQ(a, b, mtx, 16, 8);
+    return aggregate_equal_any_16x8(la, lb, mtx);
+}
+
+static int aggregate_ranges_16x8(int la, int lb, __m128i mtx[16])
```

Comment:
Both `aggregate_ranges_16x8` and `aggregate_ranges_8x16` should have prefix `_sse2neon_`.